core/src/test/scala/org/apache/spark/rpc/RpcEnvSuite.scala

```diff
@@ -624,7 +624,9 @@ abstract class RpcEnvSuite extends SparkFunSuite with BeforeAndAfterAll {
       val e = intercept[SparkException] {
         ThreadUtils.awaitResult(f, 1 seconds)
       }
-      assert(e.getCause.isInstanceOf[NotSerializableException])
+      assert(e.getCause.isInstanceOf[RuntimeException])
```

Comment:
you can see here [TransportResponseHandler.java#L207](https://github.com/apache/spark/blob/master/common/network-common/src/main/java/org/apache/spark/network/client/TransportResponseHandler.java#L207)